autogpt_platform/frontend/src/app/(platform)/library/agents/[id]/page.tsx

```diff
@@ -179,23 +197,24 @@ export default function AgentRunsPage(): React.ReactElement {
     };
   }, [api, agent?.graph_id, selectedView.id]);
 
-  // load selectedRun based on selectedView
+  // Load selectedRun based on selectedView; refresh on agent refresh
   useEffect(() => {
     if (selectedView.type != "run" || !selectedView.id || !agent) return;
 
     const newSelectedRun = agentRuns.find((run) => run.id == selectedView.id);
     if (selectedView.id !== selectedRun?.id) {
       // Pull partial data from "cache" while waiting for the rest to load
       setSelectedRun(newSelectedRun ?? null);
-
-      // Ensure corresponding graph version is available before rendering I/O
-      api
-        .getGraphExecutionInfo(agent.graph_id, selectedView.id)
-        .then(async (run) => {
-          await getGraphVersion(run.graph_id, run.graph_version);
-          setSelectedRun(run);
-        });
     }
+
+    // Load/refresh the view, even if selected view doesn't change
+    api
+      .getGraphExecutionInfo(agent.graph_id, selectedView.id)
+      .then(async (run) => {
+        // Ensure corresponding graph version is available before rendering I/O
+        await getGraphVersion(run.graph_id, run.graph_version);
+        setSelectedRun(run);
+      });
```

Comment:
Gotta check that this doesn't cause the page to spazz when receiving execution updates